coderd/insights_test.go

```diff
@@ -1204,6 +1206,9 @@ func TestTemplateInsights_Golden(t *testing.T) {
 			templates, users, testData := prepareFixtureAndTestData(t, tt.makeFixture, tt.makeTestData)
 			client := prepare(t, templates, users, testData)
 
+			// TODO(mafredri): Remove the need for this.
+			time.Sleep(3 * time.Second)
+
```

Comment:
I wonder if you could (ab)-use the locking mechanism you introduced elsewhere?
As in, start the rollup, and then try to acquire the lock. Once the lock is released, you know rollup is done.

Reply:
I rewrote this using a channel for capturing events from the `dbrollup` service.